lib/applications/resolvers.ts

```diff
@@ -611,3 +613,93 @@ export const createReplacementApplication: Resolver = async (_, args, { prisma }
     ok: true,
   };
 };
+
+export const generateApplicantsReport: Resolver = async (_, args, { prisma }) => {
+  const {
+    input: { startDate, endDate, columns },
+  } = args;
+
+  const columnsSet = new Set(columns);
+
+  const applications: GenerateApplicantsReportApplications[] = await prisma.application.findMany({
+    where: {
+      createdAt: {
+        gte: new Date(startDate),
+        lte: new Date(endDate),
+      },
+    },
+    select: {
+      rcdUserId: columnsSet.has(ApplicationsReportColumn.UserId),
+      firstName: columnsSet.has(ApplicationsReportColumn.ApplicantName),
+      middleName: columnsSet.has(ApplicationsReportColumn.ApplicantName),
+      lastName: columnsSet.has(ApplicationsReportColumn.ApplicantName),
+      dateOfBirth: columnsSet.has(ApplicationsReportColumn.ApplicantDob),
+      createdAt: columnsSet.has(ApplicationsReportColumn.ApplicationDate),
+      paymentMethod: columnsSet.has(ApplicationsReportColumn.PaymentMethod),
+      processingFee:
+        columnsSet.has(ApplicationsReportColumn.FeeAmount) ||
+        columnsSet.has(ApplicationsReportColumn.TotalAmount),
+      donationAmount:
+        columnsSet.has(ApplicationsReportColumn.DonationAmount) ||
+        columnsSet.has(ApplicationsReportColumn.TotalAmount),
+      permits: {
+        select: {
+          rcdPermitId: true,
+        },
+      },
```

Comment:
The notion ticket suggests to 

> Create an object in lib/applications/constants.ts that maps the ApplicationsReportColumn enums to strings that Prisma would consume when selecting columns in a DB query

I didn't implement that but I could extract the highlighted lines into a function if you think that makes the code cleaner

Reply:
Your current approach works fine as well! We can optimize later